engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/resolve/MethodDefinitions.scala

```diff
@@ -100,19 +100,27 @@ case object MethodDefinitions extends IRPass {
           case Some(Resolution(ResolvedType(_, tp)))
               if canGenerateStaticWrappers(tp) =>
             val dup = method.duplicate()
+            // This is the self argument that will receive the `SelfType.type` value upon dispatch, it is added to avoid modifying the dispatch mechanism.
+            val fakeModuleSelfArg = DefinitionArgument.Specified(
```

Comment:
Calling the argument `fake` is a bit scary. Maybe `synthetic`?

Reply:
Fair point, I will rename it